### PR TITLE
[GEOT-7629] document org.geotools.gce.imagemosaic.sampleimage.allowlist

### PR DESCRIPTION
**Pull request by jodygarnett**
[GEOT-7629](https://osgeo-org.atlassian.net/browse/GEOT-7629) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [ ] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Comment by jodygarnett** [comment]
@sikeoka this is a follow up to your most recent commit.

File: docs/user/library/coverage/mosaic.rst [hunk 1]
```diff
@@ -57,7 +58,18 @@ An example data set looks something like::
   
   The usual projection file associated with a shapefile, using the WKT format for a
   CoordinateReferenceSystem.
+
+* Sample image file
+
+  The :file:`sample_image.dat` (legacy :file:`sample_image`) is a sample image used to determine color model
+  upfront without a full scan. This can be useful during initial creation of an image mosiac, to witch graducule are added over time.
```
Comment (jodygarnett) [inline docs/user/library/coverage/mosaic.rst:65]:
```suggestion
  upfront without requiring a full scan.```

File: docs/user/library/coverage/mosaic.rst [hunk 2]
```diff
@@ -57,7 +58,18 @@ An example data set looks something like::
   
   The usual projection file associated with a shapefile, using the WKT format for a
   CoordinateReferenceSystem.
+
+* Sample image file
+
+  The :file:`sample_image.dat` (legacy :file:`sample_image`) is a sample image used to determine color model
+  upfront without requiring a full scan.```
```
Comment (jodygarnett) [inline docs/user/library/coverage/mosaic.rst:65]:
```suggestion
  upfront without requiring a full scan.
```

**Review by mprins — APPROVED** [review]
lgtm

**Review by sikeoka — APPROVED** [review]

**Comment by jodygarnett** [comment]
Sigh I was not aware of the back ports so going to need to change the update page appropriately.

**Pull request merged**